activesupport/lib/active_support/number_helper/number_to_rounded_converter.rb

```diff
@@ -30,7 +30,7 @@ def convert
           formatted_string =
             if BigDecimal === rounded_number && rounded_number.finite?
               s = rounded_number.to_s('F') + '0'*precision
-              a, b = s.split('.', 2)
+              a, b = s.split('.'.freeze, 2)
               a + '.' + b[0, precision]
```

Comment:
Yeah, i actually missed it. It was thr in the report too line 37 ` a + '.' + b[0, precision]`. This is also a problem and overall we have _3000 dot strings_

```
     3002  "."
      1000  /Users/agupta/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.6/lib/active_support/number_helper/number_to_delimited_converter.rb:15
      1000  /Users/agupta/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.6/lib/active_support/number_helper/number_to_rounded_converter.rb:33
      1000  /Users/agupta/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.6/lib/active_support/number_helper/number_to_rounded_converter.rb:37
```

From the couple of approaches we have 

```ruby
require 'rubygems'
require 'benchmark/ips'

# Current Implementation
def method_one
  a = 'foo'
  b = 'bar'
  a + '.' + b
end

# Appending
def method_two
  a = 'foo'
  b = 'bar'
  a << '.' << b
end

# Interpolation
def method_three
  a = 'foo'
  b = 'bar'
  a << ".#{b}"
end

Benchmark.ips do |x|
  x.config(:time => 5, :warmup => 2)
  x.report("method_one") { method_one }
  x.report("method_two") { method_two }
  x.report("method_three") { method_three }
  x.compare!
end


p "method_one output: #{method_one}"
p "method_two output: #{method_two}"
p "method_three output: #{method_three}"

```

Benchmark Output

```ruby
agupta01sjl:check_active_support agupta$ ruby bm.rb 
Warming up --------------------------------------
          method_one    96.895k i/100ms
          method_two   103.509k i/100ms
        method_three    99.381k i/100ms
Calculating -------------------------------------
          method_one      2.290M (± 4.5%) i/s -     11.434M
          method_two      2.470M (± 6.0%) i/s -     12.318M
        method_three      2.487M (± 5.3%) i/s -     12.423M

Comparison:
        method_three:  2486823.3 i/s
          method_two:  2469837.1 i/s - same-ish: difference falls within error
          method_one:  2290160.7 i/s - same-ish: difference falls within error

"method_one output: foo.bar"
"method_two output: foo.bar"
"method_three output: foo.bar"
```

*method_three* (which is interpolation) is better but it would also create a string like `.100` in this case. I am planning to go ahead with 

```ruby
a << '.'.freeze << b[0, precision]
```

Wdyt @schneems ? The above removes all the extra 3000 strings and single allocation.